docs/gateway/call-the-openai-responses-api.mdx

```diff
@@ -0,0 +1,428 @@
+
+<Step title="Call the LLM">
+
+<Note>OpenAI web search can take up to a minute to complete.</Note>
+
+```python
+response = t0.inference(
+    model_name="gpt-5-mini-responses-web-search",
+    input={
+        "messages": [
+            {
+                "role": "user",
+                "content": "What is the current population of Japan?",
+            }
+        ]
+    },
+)
+```
+
+</Step>
+
+</Steps>
+
+</Tab>
+
+<Tab title="Python (OpenAI SDK)">
+
+The TensorZero Python SDK integrates with the OpenAI Python SDK to provide access to the Responses API.
+
+<Steps>
+
+<Step title="Set up your OpenAI API key">
+
+You can set the `OPENAI_API_KEY` environment variable with your API key.
+
+```bash
+export OPENAI_API_KEY="sk-..."
+```
+
+</Step>
+
+<Step title="Install the OpenAI and TensorZero Python SDKs">
+
+You can install the OpenAI and TensorZero SDKs with a Python package manager like `pip`.
+
+```bash
+pip install openai tensorzero
+```
+
+</Step>
+
+<Step title="Configure a model for the OpenAI Responses API">
+
+Create a configuration file with a model using `api_type = "responses"` and provider tools:
+
+```toml title="tensorzero.toml"
+[models.gpt-5-mini-responses-web-search]
+routing = ["openai"]
+
+[models.gpt-5-mini-responses-web-search.providers.openai]
+type = "openai"
+model_name = "gpt-5-mini"
+api_type = "responses"
+include_encrypted_reasoning = true
+provider_tools = [{type = "web_search"}]  # built-in OpenAI web search tool
+```
+
+</Step>
+
+<Step title="Deploy a standalone (HTTP) TensorZero Gateway">
+
+Let's deploy a standalone TensorZero Gateway using Docker.
+For simplicity, we'll use the gateway with the configuration above.
+
+```bash
+docker run \
+  -e OPENAI_API_KEY \
+  -v $(pwd)/tensorzero.toml:/app/config/tensorzero.toml:ro \
+  -p 3000:3000 \
+  tensorzero/gateway \
+  --config-file /app/config/tensorzero.toml
+```
+
+<Tip>
+
+See the [TensorZero Gateway Deployment](/deployment/tensorzero-gateway) page for more details.
+
+</Tip>
+
+</Step>
+
+<Step title="Initialize the OpenAI client">
+
+Let's initialize the OpenAI SDK and point it to the gateway we just launched.
+
+```python
+from openai import OpenAI
+
+oai = OpenAI(api_key="not-used", base_url="http://localhost:3000/openai/v1")
+```
+
+<Tip>
+
+The TensorZero Python SDK supports both the synchronous `OpenAI` client and the asynchronous `AsyncOpenAI` client.
+Both options support running the gateway embedded in your application with `patch_openai_client` or connecting to a standalone gateway with `base_url`.
+See [Clients](/gateway/clients/) for more details.
+
+</Tip>
+
+</Step>
+
+<Step title="Call the LLM">
+
+<Note>OpenAI web search can take up to a minute to complete.</Note>
+
+```python
+response = oai.chat.completions.create(
+    model="tensorzero::model_name::gpt-5-mini-responses-web-search",
+    messages=[
+        {
+            "role": "user",
+            "content": "What is the current population of Japan?",
+        }
+    ],
+)
+```
```

Comment:
can we include the responses in this? I think it's unsatisfying to read this as someone who's not got a shell open.